src/openforms/js/components/admin/form_design/logic/actions/dmn/DMNActionConfig.js

```diff
@@ -24,32 +25,40 @@ const ERRORS = {
 };
 
 const DecisionDefinitionIdField = () => {
-  const {values, setFieldValue, getFieldProps, errors, touched} = useFormikContext();
+  const {
+    values: {pluginId},
+    setFieldValue,
+    getFieldProps,
+    errors,
+    touched,
+    setFieldTouched,
+    handleChange,
+  } = useFormikContext();
   const [field] = useField('decisionDefinitionId');
-  const [decisionDefinitions, setDecisionDefinitions] = useState([]);
 
-  useAsync(async () => {
-    if (!values.pluginId) {
-      setFieldValue('decisionDefinitionId', '');
-      setDecisionDefinitions([]);
-      return;
-    }
+  const {error, value: decisionDefinitions = []} = useAsync(async () => {
+    if (!pluginId) return [];
 
     const response = await get(DMN_DECISION_DEFINITIONS_LIST, {
-      engine: values.pluginId,
+      engine: pluginId,
     });
 
-    const definitionChoices = response.data.map(item => [item.id, item.label]);
-    setDecisionDefinitions(definitionChoices);
+    return response.data.map(item => [item.id, item.label]);
+  }, [pluginId, setFieldValue]);
 
-    // If pluginId has changed, reset the value of the definition
-    if (
-      values.decisionDefinitionId !== '' &&
-      !definitionChoices.find(item => item[0] === values.decisionDefinitionId)
-    ) {
-      setFieldValue('decisionDefinitionId', '');
+  useAsync(async () => {
+    if (!touched.pluginId) return;
+
+    if (!pluginId) {
+      await setFieldValue('decisionDefinitionId', '');
+      return;
     }
-  }, [values.pluginId, setFieldValue]);
+
+    // If pluginId has changed, reset the value of the definition
+    await setFieldValue('decisionDefinitionId', '');
+  }, [pluginId, setFieldValue]);
```

Comment:
What's the reason for calling these with `async`/`await`? I'd expected this to be a normal, sync `useEffect` hook and not `useAsync` :thinking:

Reply:
I dont know, the docs say that setFieldValue returns a promise, so I thought that the async would be needed?

Reply:
ah, that only really impacts what you do with the return value:

```js
foo = setFieldValue(...);
```
-> `foo` will be a promise so if you want to do something with the return value, you need to:

```js
foo = await setFieldValue(...)
// do more stuff with the return value
```

but since we don't use the return value of the (promise of) setFieldValue, it doesn't really matter. The side-effect is going to run no matter what.

I don't even know what the return value of `setFieldValue` promises looks like :D

Reply:
> If validateOnChange is set to true and there are errors, they will be resolved in the returned Promise.

right, so it returns validation errors. I don't see how that's useful for our needs, so you don't need to `await` it.